cmd/cloud_run.go

```diff
@@ -0,0 +1,381 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"go.k6.io/k6/cloudapi"
+	"go.k6.io/k6/cmd/state"
+	"go.k6.io/k6/errext"
+	"go.k6.io/k6/errext/exitcodes"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/lib/consts"
+	"go.k6.io/k6/ui/pb"
+)
+
+// cmdCloudRun handles the `k6 cloud run` sub-command
+type cmdCloudRun struct {
+	gs *state.GlobalState
+
+	showCloudLogs bool
+	exitOnRunning bool
+	uploadOnly    bool
+}
+
+func getCmdCloudRun(gs *state.GlobalState) *cobra.Command {
+	c := &cmdCloudRun{
+		gs:            gs,
+		showCloudLogs: true,
+		exitOnRunning: false,
+		uploadOnly:    false,
+	}
+
+	exampleText := getExampleText(gs, `
+  # Run a test script in Grafana Cloud k6
+  $ {{.}} cloud run script.js
+
+  # Run a test archive in Grafana Cloud k6
+  $ {{.}} cloud run archive.tar
+
+  # Read a test script or archive from stdin and run it in Grafana Cloud k6
+  $ {{.}} cloud run - < script.js`[1:])
+
+	cloudRunCmd := &cobra.Command{
+		Use:   cloudRunCommandName,
+		Short: "Run a test in Grafana Cloud k6",
+		Long: `Run a test in Grafana Cloud k6.
+
+This will execute the test in the Grafana Cloud k6 service. Using this command requires to be authenticated
+against Grafana Cloud k6. Use the "k6 cloud login" command to authenticate.`,
+		Example: exampleText,
+		Args: exactArgsWithMsg(1,
+			"the k6 cloud run command expects a single argument consisting in either a path to a script or "+
+				"archive file, or the \"-\" symbol indicating the script or archive should be read from stdin",
+		),
+		PreRunE: c.preRun,
+		RunE:    c.run,
+	}
+
+	cloudRunCmd.Flags().SortFlags = false
+	cloudRunCmd.Flags().AddFlagSet(c.flagSet())
+
+	return cloudRunCmd
+}
+
+//nolint:dupl // remove this statement once the migration from the `k6 cloud` to the `k6 cloud run` is complete.
```

Comment:
Same, why do we have this rule?

Reply:
I've modified the messages to make it clearer that it's currently duplicated code. Expect this to go away as we revamp the implementation to avoid duplication.